### DISCUSSION TITLE
Divergent branches error message

### DISCUSSION THREAD

**Discussion by thku8507** [earthlab-education/Earth-Analytics-2023-01-Intro#287] [! GitHub help] [answered]
I'm trying to sync changes to my .ipynb and .html for the Temperature Time Series Analysis project so that they show up on my github.io page, but am getting the error "Need to reconcile divergent branches" (screenshot below). 

I think @nkorinek went over how to deal with this in our discussion section but I can't remember and am having trouble interpreting what google has to say about it. Does anyone have suggestions?

![image](https://github.com/earthlab-education/Earth-Analytics-2023-01-Intro/assets/143355157/1c2503b2-ae53-421a-aabd-479cd0c6a7c9)

**Comment by jsmi374** [comment]
Hello @thku8507, 

I think I had a similar issue last week. Do you have multiple commits that you are trying to sync from jupyter notebooks? I think within the error readout in Jupyter notebooks there are some commands that you can type in to merge your divergent branches.... I used one of these commands that mentioned mentioned "merge" and "rebase from main". I forget what exactly the command was, but this solved my issue last week, that sounds similar to yours. Sorry, if I am off on this!

**Comment by eculler** [comment]
Hi @thku8507 , you need to configure git to use one of the methods or reconciling differences (doesn't matter which one; I recommend the *merge* method). In your terminal, run:

`git config --global pull.rebase false`

Then you should be able to sync.

A tip about git - it often gives you the solution in the response. You should be able to find pretty similar code if you scroll up in your terminal. I added the `--global` tag just because in the configuration I set up for you all the default directory is not actually the git repository. An alternative would be:

```bash
cd <path/to/git/repo>
git config pull.rebase false
```

`cd` stands for "change directory" and is how you navigate in the terminal.

**Reply by thku8507** [reply]
Thank you @eculler and @jsmi374! Running that in my terminal solved the issue.

**Comment by peterkobylarz** [comment]
@eculler I'm running into a sync issue that I'm not able to resolve. The command you mentioned above goes through the terminal, but I'm still getting this error:

fatal: Not possible to fast-forward, aborting.

Any suggestions? Thanks!

**Reply by peterkobylarz** [reply]
I'm also getting this error when trying various commands: not a git repository (or any of the parent directories): .git

**Reply by eculler** [reply, marked as answer]
Can we try debugging this on zoom? I haven't seen those errors before, but I'm guessing something has gone wonky with your config command. For one this, fast-forwarding is a direct alternative to `pull.rebase false`. 

The reason I added the `--global` keyword is because the default working directory in the Codespace you are using is not the repository directory, so it doesn't surprise me you are unable to run git commands without changing your directory. I don't think I'd recommend blindly running git commands, though - you can get in a real pickle. If you are really stuck, I actually recommend cloning the repo from scratch and then copying your changes in if you can't resolve conflicts with commands you know...we can go through the options when we meet.
